compiler/src/iree/compiler/ConstEval/Runtime.cpp

```diff
@@ -149,13 +141,19 @@ void CompiledBinary::deinitialize() {
 
 FunctionCall::FunctionCall(CompiledBinary &binary, iree_host_size_t argCapacity,
                            iree_host_size_t resultCapacity)
-    : binary(binary) {
-  IREE_CHECK_OK(iree_vm_list_create(iree_vm_make_undefined_type_def(),
-                                    argCapacity, iree_allocator_system(),
-                                    &inputs));
-  IREE_CHECK_OK(iree_vm_list_create(iree_vm_make_undefined_type_def(),
-                                    resultCapacity, iree_allocator_system(),
-                                    &outputs));
+    : binary(binary), argCapacity(argCapacity), resultCapacity(resultCapacity) {
+}
+
+LogicalResult FunctionCall::initialize() {
+  if (!iree_status_is_ok(iree_vm_list_create(iree_vm_make_undefined_type_def(),
+                                             argCapacity,
+                                             iree_allocator_system(), &inputs)))
+    return failure();
+  if (!iree_status_is_ok(
+          iree_vm_list_create(iree_vm_make_undefined_type_def(), resultCapacity,
+                              iree_allocator_system(), &outputs)))
```

Comment:
these should probably be default nowadays? or better, properly captured so we have only a single place that decides and the rest are `iree_allocator_t host_allocator` members or arguments

Reply:
https://github.com/openxla/iree/pull/16812 is still pending